### ISSUE TITLE
Mocha Instalin Error: ... could not spawn `g++ -fPIC -Wall -O3 -shared -fopenmp -o libmochaext.so im2col.cpp pooling.cpp`: 

### ISSUE THREAD

**Issue by paulanalyst** [pluskid/Mocha.jl#50]
Pkg.test("Mocha") is on the end...
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "help()" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.3.3 (2014-11-23 20:19 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-w64-mingw32

julia> Pkg.add("Mocha")
WARNING: julia is fixed at 0.3.3 conflicting with requirement for LinearAlgebra: [0.4.0-,?)
INFO: Cloning cache of Logging from git://github.com/kmsquire/Logging.jl.git
INFO: Cloning cache of Mocha from git://github.com/pluskid/Mocha.jl.git
INFO: Installing Logging v0.0.5
INFO: Installing Mocha v0.0.6
INFO: Building Blosc
INFO: Building LibCURL
INFO: Building WinRPM
INFO: Downloading http://download.opensuse.org/repositories/windows:/mingw:/win32/openSUSE_13.1//repodata/repomd.x
ml
INFO: Downloading http://download.opensuse.org/repositories/windows:/mingw:/win64/openSUSE_13.1//repodata/repomd.x
ml
INFO: Building HDF5
INFO: Building Mocha
Running `g++ -fPIC -Wall -O3 -shared -fopenmp -o libmochaext.so im2col.cpp pooling.cpp`
=================================================[ ERROR: Mocha ]=================================================

could not spawn `g++ -fPIC -Wall -O3 -shared -fopenmp -o libmochaext.so im2col.cpp pooling.cpp`: no such file or d
irectory (ENOENT)
while loading C:\Users\SAMSUNG2\.julia\v0.3\Mocha\deps\build.jl, in expression starting on line 23

==================================================================================================================

=================================================[ BUILD ERRORS ]=================================================

WARNING: Mocha had build errors.

 - packages with build errors remain installed in C:\Users\SAMSUNG2\.julia\v0.3
 - build a package and all its dependencies with `Pkg.build(pkg)`
 - build a single package by running its `deps/build.jl` script

==================================================================================================================

INFO: Package database updated

julia> using Mocha
24-sty 16:28:58:INFO:root:Configuring Mocha...
24-sty 16:28:58:INFO:root: * CUDA       disabled by default
24-sty 16:28:58:INFO:root: * Native Ext disabled by default
24-sty 16:28:58:INFO:root:Mocha configured, continue loading module...
Warning: using Mocha.Layer in module Main conflicts with an existing identifier.
Warning: using Mocha.dec in module Main conflicts with an existing identifier.

julia> Pkg.update()
INFO: Updating METADATA...
INFO: Updating cache of MinimalPerfectHashes...
INFO: Updating cache of MinimalPerfectHashes...
INFO: Updating LinearAlgebra...
INFO: Updating FLANN...
INFO: Updating Distributions...
INFO: Computing changes...
WARNING: julia is fixed at 0.3.3 conflicting with requirement for LinearAlgebra: [0.4.0-,?)
INFO: No packages to install, update or remove

julia> Pkg.build("Mocha")
INFO: Building Blosc
INFO: Building LibCURL
INFO: Building WinRPM
INFO: Downloading http://download.opensuse.org/repositories/windows:/mingw:/win32/openSUSE_13.1//repodata/repomd.x
ml
INFO: Downloading http://download.opensuse.org/repositories/windows:/mingw:/win64/openSUSE_13.1//repodata/repomd.x
ml
INFO: Building HDF5
INFO: Building Mocha
Running `g++ -fPIC -Wall -O3 -shared -fopenmp -o libmochaext.so im2col.cpp pooling.cpp`
=================================================[ ERROR: Mocha ]=================================================

could not spawn `g++ -fPIC -Wall -O3 -shared -fopenmp -o libmochaext.so im2col.cpp pooling.cpp`: no such file or d
irectory (ENOENT)
while loading C:\Users\SAMSUNG2\.julia\v0.3\Mocha\deps\build.jl, in expression starting on line 23

==================================================================================================================

=================================================[ BUILD ERRORS ]=================================================

julia> Pkg.test("Mocha")
INFO: Testing Mocha
24-sty 16:36:31:INFO:root:Configuring Mocha...
24-sty 16:36:31:INFO:root: * CUDA       disabled by default
24-sty 16:36:31:INFO:root: * Native Ext disabled by default
24-sty 16:36:31:INFO:root:Mocha configured, continue loading module...
-- Testing network topology with duplicated blobs
24-sty 16:36:38:INFO:root:Constructing net net on CPUBackend...
24-sty 16:36:38:INFO:root:Topological sorting 1 layers...
24-sty 16:36:38:INFO:root:Constructing net net on CPUBackend...
24-sty 16:36:38:INFO:root:Topological sorting 2 layers...
-- Testing network topology with missing blobs
24-sty 16:36:38:INFO:root:Constructing net net on CPUBackend...
24-sty 16:36:38:INFO:root:Topological sorting 1 layers...
-- Testing network topology with circular dependency
24-sty 16:36:38:INFO:root:Constructing net net on CPUBackend...
24-sty 16:36:38:INFO:root:Topological sorting 2 layers...
-- Testing network topology with multiple back-propagate path
    > Good blob sharing
24-sty 16:36:38:INFO:root:Constructing net net on CPUBackend...
24-sty 16:36:38:INFO:root:Topological sorting 5 layers...
24-sty 16:36:38:INFO:root:Setup layers...
24-sty 16:36:39:INFO:root:Network constructed!
24-sty 16:36:40:DEBUG:root:Destroying network net
    > Bad blob sharing
24-sty 16:36:40:INFO:root:Constructing net net on CPUBackend...
24-sty 16:36:40:INFO:root:Topological sorting 6 layers...
24-sty 16:36:40:INFO:root:Setup layers...
24-sty 16:36:40:INFO:root:Network constructed!
-- Testing network topology with dangling blob
    > Good case
24-sty 16:36:40:INFO:root:Constructing net net on CPUBackend...
24-sty 16:36:40:INFO:root:Topological sorting 4 layers...
24-sty 16:36:40:INFO:root:Setup layers...
ERROR: HDF5OutputLayer: output file 'C:\Users\SAMSUNG2\AppData\Local\Temp\julF77E.tmp' already exists
 in error at error.jl:21
while loading C:\Users\SAMSUNG2\.julia\v0.3\Mocha\test\net/topology.jl, in expression starting on line 112
while loading C:\Users\SAMSUNG2\.julia\v0.3\Mocha\test\runtests.jl, in expression starting on line 33

=================================================[ ERROR: Mocha ]=================================================

failed process: Process(`'C:\Users\SAMSUNG2\AppData\Local\Julia-0.3.3\bin/julia' 'C:\Users\SAMSUNG2\.julia\v0.3\Mo
cha\test\runtests.jl'`, ProcessExited(1)) [1]

==================================================================================================================

WARNING: Mocha had build errors.

 - packages with build errors remain installed in C:\Users\SAMSUNG2\.julia\v0.3
 - build a package and all its dependencies with `Pkg.build(pkg)`
 - build a single package by running its `deps/build.jl` script

==================================================================================================================

julia> using Mocha

julia> data  = HDF5DataLayer(name="train-data",source="train-data-list.txt",batch_size=64)
HDF5DataLayer(train-data)

julia>

julia> pool  = PoolingLayer(name="pool1",kernel=(2,2),stride=(2,2),bottoms=[:conv],tops=[:pool])
PoolingLayer(pool1)

julia> conv2 = ConvolutionLayer(name="conv2",n_filter=50,kernel=(5,5),bottoms=[:pool],tops=[:conv2])
Warning: imported binding for conv2 overwritten in module Main
ConvolutionLayer(conv2)

julia> pool2 = PoolingLayer(name="pool2",kernel=(2,2),stride=(2,2),bottoms=[:conv2],tops=[:pool2])
PoolingLayer(pool2)

julia> fc1   = InnerProductLayer(name="ip1",output_dim=500,neuron=Neurons.ReLU(),bottoms=[:pool2],
                                 tops=[:ip1])
InnerProductLayer(ip1)

julia> fc2   = InnerProductLayer(name="ip2",output_dim=10,bottoms=[:ip1],tops=[:ip2])
InnerProductLayer(ip2)

julia> loss  = SoftmaxLossLayer(name="loss",bottoms=[:ip2,:label])
SoftmaxLossLayer(loss)

julia>

julia> backend = GPUBackend()
ERROR: GPUBackend not defined

julia> init(backend)
ERROR: backend not defined

Paul

**Comment by pluskid** [comment]
@paulanalyst There is no need to run `Pkg.build("Mocha")` if you plan to use pure-Julia backend. On the other hand, the GPUBackend do need a GPU device and extra configuration to run. Please see the [documents on CUDA backend](http://mochajl.readthedocs.org/en/latest/user-guide/backend.html#cuda-backend).

**Comment by n-p** [comment]
You should change slashes to backslashes in windows.
